src/com/owncloud/android/ui/preview/PreviewImageFragment.java

```diff
@@ -322,6 +325,10 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 mContainerActivity.getFileOperationsHelper().toggleFavorite(getFile(), false);
                 return true;
             }
+            case R.id.action_set_as_wallpaper:{
+                mContainerActivity.getFileOperationsHelper().setPictureAs(getFile());
+                return true;
+            }
```

Comment:
This will only work for `PreviewImageFragment` view, so if you want to keep it visible in file listview you can do the same in`OCFileListFragment.onFileActionChosen()`